website/server/controllers/api-v3/quests.js

```diff
@@ -193,12 +193,14 @@ api.acceptQuest = {
     if (group.quest.active) throw new NotAuthorized(res.t('questAlreadyStartedFriendly'));
     if (group.quest.members[user._id]) throw new BadRequest(res.t('questAlreadyAccepted'));
 
+    const acceptedSuccessfully = await group.handleQuestInvitation(user, true);
+    if (!acceptedSuccessfully) {
+      throw new NotAuthorized(res.t('questAlreadyAccepted'));
+    }
+
     user.party.quest.RSVPNeeded = false;
     await user.save();
 
-    group.markModified('quest');
-    group.quest.members[user._id] = true;
```

Comment:
even though it's not used anymore, could we keep this line? And the similar one for `api.rejectQuest`? Because the quest data is returned from the API call and we should return the updated state

Reply:
It's already handled inside the `group.handleQuestInvitation` method.

Reply:
I had missed that :)